src/TriggerBinding/SqlTableChangeMonitor.cs

```diff
@@ -595,7 +578,7 @@ private SqlCommand BuildGetChangesCommand(SqlConnection connection, SqlTransacti
                     c.SYS_CHANGE_VERSION, c.SYS_CHANGE_OPERATION,
                     w.ChangeVersion, w.AttemptCount, w.LeaseExpirationTime
                 FROM CHANGETABLE(CHANGES {this._userTable.BracketQuotedFullName}, @last_sync_version) AS c
-                LEFT OUTER JOIN {this._workerTableName} AS w WITH (TABLOCKX) ON {workerTableJoinCondition}
+                LEFT OUTER JOIN {this._workerTableName} AS w ON {workerTableJoinCondition}
```

Comment:
why is this updated?

Reply:
Hi, after this change, I had chance to develop more understanding on the use of appropriate isolation level and locking modes, so I may need to update the lock modes further in this file. To get to some history, there were known deadlock issues in the SQL trigger support when our team took the ownership. Following my suggestion, Ameya had added table-level exclusive locks just to make sure that there are no more deadlocks. The next task was to selectively reduce the degree or locking and/or increase locking granularity and make sure the triggers are still deadlock-free. But this was not done then so I am trying to get it done this time.

Particularly in the above case, I did not find the use of taking table-level exclusive lock as this is a SELECT statement with no follow-up INSERT/UPDATE statement. After the change to lock modes, I tested with 3 worker applications running together and could not reproduce deadlocks.

Reply:
I will be pushing the change to update the locking scheme in a future PR. This PR is closed, and the changes are separated into multiple PRs.